src/main/resource-applier.ts

```diff
@@ -21,56 +21,73 @@
 
 import type { Cluster } from "./cluster";
 import type { KubernetesObject } from "@kubernetes/client-node";
-import { exec } from "child_process";
-import fs from "fs";
+import fse from "fs-extra";
 import * as yaml from "js-yaml";
 import path from "path";
 import * as tempy from "tempy";
 import logger from "./logger";
 import { appEventBus } from "../common/event-bus";
 import { cloneJsonObject } from "../common/utils";
+import { inspect } from "util";
+import { promiseExec } from "./promise-exec";
+
+function sanitize(raw: string): string {
+  return inspect(raw, false, undefined, false);
+}
+
+function sanitizeObject(resource: KubernetesObject | any) {
```

Comment:
How is this refactoring related to PR?

Reply:
I considered this to be mostly a refactoring and correctness PR. But I realize that the PR title does not capture that. Will update the title.

Reply:
Ok, so in the end we just moved a method from a class to a separate function. Is there a reason?

Reply:
I moved it to a separate function because it doesn't need anything from the class